### ISSUE TITLE
Support Android 11 (API 30)

### ISSUE THREAD

**Issue by sgrimault** [PnX-SI/gn_mobile_occtax#88]
L'application "Occtax" n'est pas compatible avec la dernière version d'Android (11, API 30). Elle dépend des modules suivants qui sont eux mêmes non compatibles :
* [gn_mobile_maps](https://github.com/PnX-SI/gn_mobile_occtax) (cf. https://github.com/PnX-SI/gn_mobile_maps/issues/7)
* [gn_mobile_core](https://github.com/PnX-SI/gn_mobile_core) (cf. https://github.com/PnX-SI/gn_mobile_core/issues/36)

**Comment by lepontois** [comment]
Est-ce que ça explique un problème d'affichage des fonds de carte ?

**Comment by camillemonchicourt** [comment]
Oui, avec Android 11 on ne peut plus accéder aux données (carto notamment) à l'endroit et à la manière dont on le fait actuellement.

**Comment by lepontois** [comment]
Ok, c'est donc pour ça que je ne vois pas mes fond de carte.
Des développements sont-ils en cours ou programmés pour corriger ça ?

**Comment by camillemonchicourt** [comment]
Il faut les financer, les organiser puis les suivre. Le développeur a fait la première analyse technique.

**Comment by lepontois** [comment]
Par curiosité, je suis preneur de cette analyse.

**Comment by camillemonchicourt** [comment]
Elle est dans ce sujet dans le premier message, les liens vers 2 autres tickets (Sync et Maps)

**Comment by lepontois** [comment]
Autant pour moi, je pensais qu'il y avait quelque chose de plus complet.

**Comment by camillemonchicourt** [comment]
Oui tu as raison, on a eu des échanges complémentaires, assez denses.
Les voici : [2021-05-occtax-mobile-android11.txt](https://github.com/PnX-SI/gn_mobile_occtax/files/6591637/2021-05-occtax-mobile-android11.txt)

Bon courage ! :-)

**Comment by aferchal** [comment]
Je prends la discussion en route et ces problématiques de compatibilité de version vont rapidement nous poser pb. Nous sommes en train de déployer et de former les agents sur GN, d'abord sur l'interface web et j'avais validé ce passage vers GN à la place d'ObsOcc justement pour l'apport du mobile qui est très attendu. A l'heure actuelle, nous sommes avec des XCover 4S qui sont en 10 et qui étonnament ne proposent pas la maj encore. On ne peut pas la bloquer via notre outil de gestion de flotte (DME de chez Orange) et il faut le faire poste par poste ce qui prend du temps et ne convient pas forcément à notre admin réseau. De mon point de vue il faudrait rapidement se mettre en ordre de marche pour prévoir une presta sur ce sujet car on va vite se retrouver avec une appli inopérante. A voir avec l'OFB peut etre ?

**Comment by samuelpriou** [comment]
Je partage également la réflexion d'Alain @aferchal. Les agents au PNM sont équipés de smartphones XCOVER 4 (Android 9) et XCOVER 4S (Android 10). Actuellement, aucune mise à jour n'a été demandée. Nous avons pris le parti de ne pas bloquer les smartphones d'où la difficulté de bloquer les mises à jour et notamment celle  d'Android. Je suis également d'avis à prévoir une prestation sur le sujet sans quoi Occtax mobile deviendra inopérant.

**Comment by ClaireLagaye** [comment]
Nous avons au PNV 8 mobiles sur 40 qui ont été mis à jour sur Android 11 et ce malgré nos directives. Il est en fait impossible de bloquer la proposition des mises à jour à l'utilisateur, on ne peut que bloquer le téléchargement automatique.
Il faut noter aussi que pour nos A50 et A40 (je ne sais pas si c'est le cas pour tous les samsung) la mise à jour vers Android 11 est irréversible: même un hard reset ne remet pas Android 10, et rien n'est prévu vôté Samsung pour permettre cette redescente.
La seule façon que j'ai trouvé pour le faire est de redéployer via Odin un OS trouvé sur internet mais ça peut être risqué et ça fait apparemment sauter la garantie.
Nous sommes donc bien motivés pour prévoir ce développement supplémentaire et nous allons essayer de trouver du budget pour le faire dès la fin de l'été.

**Comment by camillemonchicourt** [comment]
Pour commencer, on propose une solution rapide ajoutant une permission à valider par l'utilisateur au premier lancement permettant de donner à l'application des droits d'accès à tout le terminal.
Dans un second temps on cherche une solution plus pérenne et plus conforme aux contraintes d'Android, si possible en gardant la possibilité de mutualiser les fonds de carte offline avec d'autres applications, et de pouvoir stocker les cartes offline sur le stockage interne ou externe.

**Comment by DonovanMaillard** [comment]
Bonjour, pour info sur la suite de ce sujet :

Sébastien a proposé les apk de test suivantes : 
• "Sync" : https://e.pcloud.link/publink/show?code=XZ4YaVZWmrHSVhlaupOfgNFS4MoWhpFjoDk
• "Occtax" : https://e.pcloud.link/publink/show?code=XZLYaVZVaANo657wQuFq54SkI0RXy5x0pGk\

J’ai testé dans un premier temps le déploiement sur un Android 10 (pourrécupérer le numéro de version, voir si tout fonctionne bien sur A10 avant de passer sur A11, et voir la rétrocompatibilité). 

Sauf erreur de ma part, les numéros de version pour les 

- pour occtax : 2285
- pour sync : 3140

Ensuite, j’ai supprimé mes versions précédentes, installé les nouvelles APK, la connexion ne tient pas : La connexion semble se faire dans un premier temps et permet à Sync de récupérer observateurs, nomenclatures, taxons etc : toutes les routes sans CRUVED. Mais je ne récupère pas les données protégées (JDD) et la connexion n'est pas active.

J’ai vu passer les tickets de Xavier et Claire sur le sujet, j’ai supprimé les caches des deux applis, et les données des applications en vue de pouvoir me connecter : ça ne change rien dans mon cas. Pas possible d’aller plus loin à l’heure actuelle pour tester la suite et le stockage des fonds de carte. 

Je vais reprendre la manip pour récupérer les logs sur le mobile.

**Comment by DonovanMaillard** [comment]
Suite à ces tests, nous sommes revenus aux versions précédentes (1.2.1 et 1.2.4) pour refaire des saisies. Le soucis sur les jeux de données non récupérés persiste de notre coté. 

Est-ce que d'autres ont observé le même soucis? A voir s'il faut un ticket à part, ou si c'est lié à cette montée de version + rétropédalage ou si c'était déjà présent depuis le passage à GeoNature 2.7 par exemple ?
Donovan

**Comment by camillemonchicourt** [comment]
Je pense que pour un autre sujet, il faut un autre ticket.
Le retour à un version précédente ne me paraît pas lié. A vérifier en supprimant tout et installant la 1.2.
A voir si il y a un soucis avec GN 2.7. 
Mais comme c'est pas lié à Android 11, je suggère de bouger ces 2 posts sur ce ticket dans un autre dédié. Merci.

**Comment by PNPyrenees** [comment]
Premier retour d’évaluation de cette version positif.
Étant bien sur Android V11, j'ai bien mon fond de carte qui s'affiche à nouveau. Bravo !

Par contre le problème de synchronisation de la liste de taxon persiste (mais je ne sais pas si ce problème est inclut dans cette prestation).
Détail : Lors de la synchro, il m'affiche 13747 taxons (puis il passe à la synchronisation cartographique -> 319 556) Mais sur le formulaire de saisie, le nombre de taxon disponible vari d'une synchro à l'autre (7099, 8069...). Par exemple, ma dernière synchro ne me permet pas de saisir du Gypaète Barbu... C'est embêtant.
cf : https://github.com/PnX-SI/gn_mobile_occtax/issues/81

**Comment by samuelpriou** [comment]
Bonjour à tous,
J'ai testé les deux .apk. Le smartphone de test est un Samsung A50 avec Android 11.
J'ai pu récupérer ma liste de taxons et les fonds de carte.
Merci

**Comment by DonovanMaillard** [comment]
La release 1.3.0 fonctionne sous Android 11 et précédentes.

**Comment by sgrimault** [comment]
Je la laisse ouverte car, le support pour Android 11 n'est que partiel (cf. note de version par mail sur le sujet).

**Comment by aferchal** [comment]
De retour j'ai testé la 1.3 de sync et occtax sur un XCOVER4S fraîchement mis à jour en Androïd 11 sur l'interface de recette du BRGM, ça a l'air ok. Juste un truc, l'install de SYNC 1.3.0 ne pointe pas vers la version1.3 d'OCCTAX qui est à intaller à la mano sinon il installe la 1.2.4

**Comment by DonovanMaillard** [comment]
Bonjour, 

Je n'ai pas constaté ce soucis de mon coté, la mise à jour de la version occtax a bien été faite dans la BDD (chemin apk + numéro de version?). 

Sinon Amandine avait eu un soucis potentiellement lié sur l'absence de mise à jour des applis par Sync, à creuser.

**Comment by DonovanMaillard** [comment]
> Je la laisse ouverte car le support pour Android 11 n'est que partiel (cf. note de version par mail sur le sujet).

@sgrimault  pourras tu nous préciser ici ce qu'il reste à faire pour avoir un support complet d'Android 11 au vu des évolutions récentes ou en cours de l'appli ? Merci à toi!

**Comment by sgrimault** [comment]
Tout est OK, le seul bémol concerne le module "maps" (cf. https://github.com/PnX-SI/gn_mobile_maps/issues/7) où la solution actuelle implique de passer par une permission non pérenne `MANAGE_EXTERNAL_STORAGE` pour lister les fonds de carte disponibles.

**Comment by DonovanMaillard** [comment]
@sgrimault est-ce qu'on pourra revoir ensemble quelles solutions envisager sur ce ticket,pour avoir une gestion pérenne sur les versions récentes d'Android ?

**Comment by sgrimault** [comment]
Pour l'instant on va rester sur l'utilisation de la permission `MANAGE_EXTERNAL_STORAGE` faute de mieux (cf. https://github.com/osmdroid/osmdroid/issues/1743). Il y a un sujet aussi pour faire monter de version les dépendances utilisées par le module "maps", notamment pour cibler les dernières versions d'Android.

**Comment by DonovanMaillard** [comment]
@sgrimault on peut fermer ce ticket désormais ?

**Comment by sgrimault** [comment]
Oui, on peut. On reste sur la permission `MANAGE_EXTERNAL_STORAGE` (qui risque de poser problème plus tard, surtout si on souhaite un jour passer par une distribution de l'application via Play Store).

**Comment by camillemonchicourt** [comment]
Le support Android est OK, mais le sujet de fonctionnement de la permission pour accéder à des fichiers de fonds carto n'importe où sur le terminal, n'est pas réglé et est à rediscuter pour ne pas être bloqué quand cette permission dépréciée ne sera plus supportée.

Peut-être que désormais on peut laisser tomber l'idée de pouvoir mettre les fonds carto offline où on veut et les mettre dans un dossier particulier de l'application comme souhaité par Android.

Cela pourra simplifier d'autres aspects comme le fait de plus facilement et rapidement trouver les fichiers de carto offline. 🤔

**Comment by DonovanMaillard** [comment]
Je pense qu'on peut faire un ticket dédié qui n'est plus une question de support Android 11. 

On avait cette demande de stocker des fichiers n'importe où sur le terminal pour utiliser les mêmes fonds que d'autres applis comme OruxMap, les fonds étant lourds. Mais on aurait peut être d'autres approches à privilégier maintenant : 
- les fonds online couvrent 90% des surfaces et des besoins
- On a un fichier de conf custom qui peut etre chargée sur le terminal pour surcoucher la conf serveur, donc on peut imaginer un paramètre de l'utilisateur qui définit le chemin absolu de son répertoire de fonds carto. 

C'est sans doute un sujet à part entière qui permettrait d'être un peu plus conventionnels au niveau des permissions;

**Comment by camillemonchicourt** [comment]
Je pense pas que cette solution serait pertinente dans notre contexte au PNE, mais oui, à discuter dans un sujet dédié.

**Comment by DonovanMaillard** [comment]
Dans le contexte PNE, il suffit de passer le bon chemin sur la conf serveur non? La surcouche n'est pas nécessaire. 

Les chemins sont tous les mêmes ?

**Comment by sgrimault** [comment]
Plutôt passer par une issue dédiée (et plutôt coté [module cartographique](https://github.com/PnX-SI/gn_mobile_maps)), afin de voir comment gérer autrement l'accès à l'espace de stockage et éviter peut être l'usage de la permission `MANAGE_EXTERNAL_STORAGE`.
En dehors de ça le support d'Android 11 et supérieur est déjà géré depuis quelques temps déjà :)

**Comment by DonovanMaillard** [comment]
j'ai ouvert un ticket dédié mais sur ce dépôt, j'avais gardé en tête qu'on conservait les sous-modules et dépôts dédiés pour les développements, mais qu'on centralisait les tickets ici pour une meilleure lisibilité pour tout le monde

voir #317
